packages/openapi-fetch/test/types.test.ts

```diff
@@ -280,4 +286,179 @@ describe("types", () => {
       assertType<Response>({ error: "default application/json" });
     });
   });
+
+  describe("OpenApiStatusToHttpStatus", () => {
+    test("returns numeric status code", () => {
+      assertType<OpenApiStatusToHttpStatus<200, number>>(200);
+      assertType<OpenApiStatusToHttpStatus<200, string>>(200);
+      assertType<OpenApiStatusToHttpStatus<204, string>>(204);
+
+      assertType<OpenApiStatusToHttpStatus<204, string>>(
+        // @ts-expect-error 200 is not a valid
+        200,
+      );
+      assertType<OpenApiStatusToHttpStatus<204, number>>(
+        // @ts-expect-error 200 is not a valid
+        200,
+      );
+
+      assertType<OpenApiStatusToHttpStatus<404, 200 | 204 | 206 | 404 | 500 | "default">>(404);
+    });
+
+    test("returns default response", () => {
+      type Status = OpenApiStatusToHttpStatus<"default", 200 | 204 | 206 | 404 | 500 | "default">;
```

Comment:
Would it make sense add a test case where there also is a wildcard (e.g. `"2XX"`) in the second argument?